src/azure-cli-testsdk/azure/cli/testsdk/preparers.py

```diff
@@ -182,6 +182,46 @@ def _get_resource_group(self, **kwargs):
                                                self.resource_group_parameter_name))
 
 
+# KeyVault Preparer and its shorthand decorator
+
+class KeyVaultPreparer(AbstractPreparer, SingleValueReplacer):
+    def __init__(self,  # pylint: disable=too-many-arguments
+                 name_prefix='clitest', sku='standard', location='westus',
+                 parameter_name='key_vault', resource_group_parameter_name='resource_group',
+                 skip_delete=True, dev_setting_name='AZURE_CLI_TEST_DEV_KEY_VAULT_NAME'):
+        super(KeyVaultPreparer, self).__init__(name_prefix, 24)
+        self.location = location
+        self.sku = sku
+        self.resource_group_parameter_name = resource_group_parameter_name
+        self.skip_delete = skip_delete
+        self.parameter_name = parameter_name
+
+        self.dev_setting_name = os.environ.get(dev_setting_name, None)
+
+    def create_resource(self, name, **kwargs):
+        if not self.dev_setting_name:
+            group = self._get_resource_group(**kwargs)
+            template = 'az keyvault create -n {} -g {} -l {} --sku {}'
+            execute(template.format(name, group, self.location, self.sku))
+            return {self.parameter_name: name}
+        else:
+            return {self.parameter_name: self.dev_setting_name}
+
+    def remove_resource(self, name, **kwargs):
+        if not self.skip_delete and not self.dev_setting_name:
+            group = self._get_resource_group(**kwargs)
+            execute('az keyvault delete -n {} -g {} --yes'.format(name, group))
+
+    def _get_resource_group(self, **kwargs):
+        try:
+            return kwargs.get(self.resource_group_parameter_name)
+        except KeyError:
+            template = 'To create a KeyVault a resource group is required. Please add ' \
+                       'decorator @{} in front of this KeyVault preparer.'
+            raise CliTestError(template.format(KeyVaultPreparer.__name__,
+                                               self.resource_group_parameter_name))
```

Comment:
I may have missed something obvious but `template` has one parameter but 2 params are passed to `template.format(...)`.